nym-api/nym-api-requests/src/models.rs

```diff
@@ -607,6 +607,15 @@ pub struct NymNodeDescription {
 
     // for now we only care about their ws/wss situation, nothing more
     pub mixnet_websockets: WebSockets,
+
+    pub role: NymNodeRole,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, schemars::JsonSchema)]
+#[serde(rename_all = "lowercase")]
+pub enum NymNodeRole {
```

Comment:
we already have a role enum defined: https://github.com/nymtech/nym/blob/develop/nym-api/nym-api-requests/src/nym_nodes.rs#L73 maybe we could somehow refactor it a tiny bit and reuse it here too?

Reply:
We can probably guess if its an exit gateway if its running IPR, so we can assign gateway roles off that

Reply:
Actually not really, since they could plausibly be Entry as well, @jstuczyn any ideas on inferring these roles from data available to the `nym-api`